pkg/component/gardener/apiserver/apiserver.go

```diff
@@ -183,8 +184,13 @@ func (g *gardenerAPIServer) Deploy(ctx context.Context) error {
 	}
 
 	serviceRuntime := &corev1.Service{}
-	if err := g.client.Get(ctx, client.ObjectKey{Name: serviceName, Namespace: g.namespace}, serviceRuntime); err != nil {
-		return err
+	if err := retry.UntilTimeout(ctx, 5*time.Second, 30*time.Second, func(ctx context.Context) (bool, error) {
+		if err := g.client.Get(ctx, client.ObjectKey{Name: serviceName, Namespace: g.namespace}, serviceRuntime); err != nil {
+			return retry.MinorError(fmt.Errorf("service gardener-apiserver was not yet created"))
```

Comment:
Shouldn't we return a MinorError if the error type is NotFound, and a SevereError if the error is something else?